endpoint/endpoint.go

```diff
@@ -191,10 +192,52 @@ func (e *Endpoint) GetProviderSpecificProperty(key string) (ProviderSpecificProp
 	return ProviderSpecificProperty{}, false
 }
 
+type NormalizeProviderSpecificConfig map[string]func(string) string
+
+// Selects and normalizes fields specified by config
```

Comment:
Please always try to respect the godoc convention. This should start with `NormalizeProviderSpecific` and describe what the method is actually doing.